appengine/js/js-blocks.js

```diff
@@ -192,6 +197,75 @@ Blockly.Blocks['controls_whileUntil'].init = function() {
 };
 
 /**
+ * Block for 'for' loop.
+ * @this Blockly.Block
+ */
+Blockly.Blocks['controls_for'].init = function() {
+  this.jsonInit({
+    "message0": "for (%1 = %2;  ? < %3;  ? += 1) { %4 %5 }",
+    "args0": [
+      {
+        "type": "field_variable",
+        "name": "VAR",
+        "variable": null
+      },
+      {
+        "type": "input_value",
+        "name": "FROM",
+        "check": "Number",
+        "align": "RIGHT"
+      },
+      {
+        "type": "input_value",
+        "name": "TO",
+        "check": "Number",
+        "align": "RIGHT"
+      },
+      {
+        "type": "input_dummy"
+      },
+      {
+        "type": "input_statement",
+        "name": "DO"
+      }
+
+    ],
+    "inputsInline": true,
+    "previousStatement": null,
+    "nextStatement": null,
+    "colour": Blockly.Blocks.loops.HUE,
+    "helpUrl": Blockly.Msg.CONTROLS_FOR_HELPURL
+  });
+  // Assign 'this' to a variable for use in the tooltip closure below.
+  var thisBlock = this;
+  this.setTooltip(function() {
+    return Blockly.Msg.CONTROLS_FOR_TOOLTIP.replace('%1',
+        thisBlock.getFieldValue('VAR'));
+  });
+};
+
+Blockly.JavaScript['controls_for'] = function(block) {
+  // For loop.
+  var variable0 = Blockly.JavaScript.variableDB_.getName(
+      block.getFieldValue('VAR'), Blockly.Variables.NAME_TYPE);
+  var argument0 = Blockly.JavaScript.valueToCode(block, 'FROM',
```

Comment:
Optional: consider calling these argumentFrom and argumentTo? Seems easier to remember what they're referring to that way.